autotest/gdrivers/hdf5.py

```diff
@@ -1274,6 +1274,12 @@ def test_hdf5_eos_swath_no_explicit_dimension_map():
     assert gdal.Open(ds.GetMetadataItem("Y_DATASET", "GEOLOCATION")) is not None
     ds = None
 
+    ds = gdal.Open(
+        'HDF5:"data/hdf5/dummy_HDFEOS_swath.h5"://HDFEOS/SWATHS/MySwath/Geolocation_Fields/Longitude'
+    )
+    assert len(ds.GetGCPs()) == 0
+    assert ds.GetMetadata("GEOLOCATION") is not None
```

Comment:
Does it make sense to check the value of the metadata instead of it's mere existence?

Reply:
given that the fill of the GEOLOCATION metadata items of Geolocation_Fields is done in the same code path as DataFields which is tested by an upper test, I don't think that's necessary